src/server/status.rs

```diff
@@ -224,6 +242,101 @@ impl Status {
     }
 }
 
+impl RemoteStatus {
+    pub async fn new(name: String) -> anyhow::Result<Self> {
+        let deadline = Instant::now() + Duration::from_secs(1);
+        let creds_path = credentials::path(&name)?;
+        let builder = future::timeout(
+            Duration::from_secs(1), Builder::read_credentials(creds_path)
+        ).await??;
+        if let Some((host, port)) = builder.get_addr().get_tcp_addr() {
+            let mut version = None;
+            let mut status = "unknown";
+            let mut error = None;
+            match future::timeout(
+                deadline - Instant::now() , builder.connect()
+            ).await {
+                Ok(Ok(mut con)) => {
+                    if let Ok(Ok(ver)) = future::timeout(
+                        deadline - Instant::now() , con.get_version()
+                    ).await {
+                        version = Some(ver);
+                        status = "running";
+                    }
+                }
+                Ok(Err(e)) => {
+                    status = "error";
+                    error = Some(format!("{}", e));
+                }
+                Err(_) => {}  // timeout, ignore
```

Comment:
Do `log::info` for the error in the both cases. And why not `error = Some("timeout happened")` in this case?